tests/pipelines/pipeline/shared/test_utility.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+import os
+
+from dpypelines.pipeline.shared.utility import enrich_online
+from dpypelines.pipeline.shared.message import unexpected_error
+
+
+@enrich_online
+def very_much_expected_error(msg: str, error: Exception) -> str:
+    """
+    We've caught an unexpected error. Make a sensible message explaining
+    the problem.
+    """
+    error_type = error.__class__.__name__
+    message = f"""
+        {msg}
+
+        Error type: {error_type}
+        Error: {error}
+    """
+
+    return message
+
+
+def test_enrich_message():
+    os.environ["ENRICH_MESSAGE"] = "Please work"
```

Comment:
so this will set an environment variable on the users machine until they next shut it down, needs to just be for the length of this test.

see here: https://github.com/ONSdigital/dp-python-tools/blob/b256a30a4c8d4acf4584b336f22b642c8aa933a4/tests/config/test_config.py#L16 for an example of how to set it for the length of the test only.

Reply:
it doesn't need to be an exception, fundementally your decorator wraps something that returns a sting and _sometimes_ (if an env var is set) adds a bit to the end, thats all,, so we only need something very lightweight here.

```python

def test_enrich_decorator():
    """
    Test we can control the behaviour of the enrich online decorator via an env var.
    """

    @enrich_online
    def _foo_funct() -> str:
          return "foo"
          
     # no env var set
     assert _foo_func() == "foo"
     
     # TODO - set env var with monkeypatch
     
     # env var is set
     assert _foo_func() == "foo <whatever should be added"
```